HorizonCore/src/main/java/com/hubspot/horizon/HttpConfig.java

```diff
@@ -63,10 +63,37 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
+  /**
+   * The HttpClient will use this setting to throttle requests.
+   *
+   * If the client gets a request, but the max connections are already in use,
+   * the client will block indefinitely until it has a free connection for the request,
+   * then make the request once it's able.
+   *
+   * The default value is 100.
+   */
   public int getMaxConnections() {
     return maxConnections;
   }
 
+  /**
+   * Unlike maxConnections, the HttpClient will use this per-host setting to IMMEDIATELY FAIL any requests
```

Comment:
nit: should these variable and class references be javadoc links, eg `{@link #getMaxConnections}` or w/e? That way people can easily browse the code from this comment.

Reply:
sure I can do that

Reply:
oh second thought, would it be ok to merge without it? I haven't been able to get back to this (I got locked out of public GH for a month between writing these comments and opening the PR), and this is at least better than having no comments